File: compile_mustache.py
#!/usr/bin/env python
"""Render a Mustache template, optionally populated with JSON data"""


import pystache
import json
import sys


class partials(object):
    partials_root = 'templates/partials'
    def get(self, name):
        """return the template code of a partial located
        in templates/partials/
        """
        r = ""
        try:
            with open(self.partials_root+'/'+name+'.mustache') as partial:
                r = partial.read().decode('utf-8')
        except IOError:
            sys.stderr.write('Couldn\'t find partial '+name+'\n')
        return r


try:
    template = open(sys.argv[1]).read().decode('utf-8')
except Exception:
    sys.stderr.write("usage: compile_templates template.mustache [data.json]\n")
    sys.exit(1)

try:
    view = json.load(open(sys.argv[2]))
except:
    view = {}

renderer = pystache.Renderer(partials=partials())
sys.stdout.write(renderer.render(template, view).encode('utf-8'))

File: compile_mustache.js
#!/usr/bin/env node
/**
 * Render a Mustache template, optionally populated with JSON data
 */


var Mustache = require('mustache');
var fs = require('fs');
var view = {};
var partials_root = 'templates/partials';

try {
  var template = fs.readFileSync(process.argv[2], 'utf-8');
} catch (e) {
  process.stderr.write("usage: compile_templates template.mustache [data.json]\n");
  process.exit(1);
}

try {
  view = JSON.parse(fs.readFileSync(process.argv[3], 'utf-8'));
} catch (e) {}

/**
 * return the template code of a partial located in templates/partials/
 */
function partials(name) {
  var template = '';
  try {
    template = fs.readFileSync(partials_root+'/'+name+'.mustache',
                               'utf-8');
  } catch (e) {
    process.stderr.write('Couldn\'t find partial '+name+'\n');
  }
  return template;
}


var output = Mustache.render(template, view, partials);
process.stdout.write(output);
